File: examples/python/advanced/error-handling/validation-error.py
from aws_durable_execution_sdk_python import (
    DurableContext,
    durable_execution,
    ValidationError,
)
from aws_durable_execution_sdk_python.config import CallbackConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Handle SDK validation errors."""
    try:
        # SDK raises ValidationError if timeout is invalid
        callback = context.create_callback(
            config=CallbackConfig(timeout_seconds=-1),  # Invalid!
            name="approval"
        )
        return {"callback_id": callback}
    except ValidationError as e:
        # SDK caught invalid configuration
        return {"error": "InvalidConfiguration", "message": str(e)}

File: examples/java/advanced/error-handling/validation-error.java
try {
    // SDK raises exception if configuration is invalid
    var callback = ctx.createCallback("approval", String.class,
        CallbackConfig.builder()
            .timeout(Duration.ofSeconds(-1))  // Invalid!
            .build());
    return Map.of("callback_id", callback.callbackId());
} catch (IllegalArgumentException e) {
    // SDK caught invalid configuration
    return Map.of("error", "InvalidConfiguration", "message", e.getMessage());
}
